scripts/core/exp-command-buffer.yml

```diff
@@ -170,6 +170,11 @@ desc: "Descriptor type for updating a kernel launch command."
 base: $x_base_desc_t
 name: $x_exp_command_buffer_update_kernel_launch_desc_t
 members:
+    - type: $x_kernel_handle_t
+      name: hNewKernel
+      desc: |
+        [in] The new kernel handle. If this value is equal to the current kernel handle associated 
```

Comment:
It doesn't look like `hNewKernel` is optional. I'm not sure if we discussed whether it should be to allow devices that don't support kernel update to still be able to update other things.

The  `COMMAND_BUFFER_UPDATE_SUPPORT_EXP`   isn't fine-grained enough to represent individual aspects of mutability. Technically in this PR every adapter other than CUDA could report false for this now as they don't support kernel binary update, but I'm not sure if we actually want to do that as it will reduce test coverage for other aspects of update.

Reply:
I think the reduction in test coverage would be a problem. I think that if we are going to have an implementation for other adapters soon, it would make sense to not change the aspect. 

I'm curious about why OpenCL is hardcoded to false? There seems to be an implementation of urCommandBufferUpdateKernelLaunchExp in the opencl adapter

Reply:
Regarding `hNewKernel` not being optional: I wasn't sure whether to make it optional or not when writing the spec.

 In the end I decided to make it non-optional and require the user to explicitly pass the kernel handle that is currenty active if they only want to change the parameters. To me passing the kernel handle instead of nullptr seems a bit more consistent but happy to discuss this.

Reply:
I went back on my opinion about this and decided to keep `kNewKernel` but make it optional (as Ewan had suggested before). It just seemed more consistent after the recent change to make the api more flexible with less error conditions.

Note: `hNewKernel` is not a pointer to a handle since it seems to be a common pattern UR to have optional handles that are not marked as pointers. For example: `urMemGetNativeHandle` and `urUSMHostAlloc`